commons/src/main/java/org/infinispan/commons/configuration/attributes/AttributeCopier.java

```diff
@@ -0,0 +1,13 @@
+package org.infinispan.commons.configuration.attributes;
+
+/**
+ * AttributeCopier. Usually an attribute is copied by using the {@link Object#clone()} method. When
+ * this method is not enough, you can provide a custom attribute copier by implementing this
+ * interface
+ *
+ * @author Tristan Tarrant
+ * @since 7.2
+ */
+public interface AttributeCopier {
```

Comment:
I personally feel typing this to T on the class would make it a bit cleaner.  You currently type the AttributeDefinition itself, so why not the copier?  Then I would suggest breaking out the CollectionAttributeCopier into 2 classes.  One that is actually just for java.util.Collection classes and the other for TypedProperties.  From its name I would not guess that TypedProperties could be used with it, and unfortunately the compiler doesn't help either without defined types.